### ISSUE TITLE
perf(agent): skip redundant same-tenant exact L1 snapshot readmission

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#11964]
Parent: #11572; hardware packet: #11959. Follow-on to #9527; existing #9537-#9539 cover broader QA, not same-tenant scope optimization.

For concurrent agents using one authenticated tenant on fak-native serving, exact device-L1 prefix hits currently discard the lookup source scope and capture/readmit the unchanged complete snapshot. Preserve the returned source scope and skip readmission when the exact cached-logits L1 hit already came from ScopeTenant, the same destination used by AdmitPrivateSnapshot. Also keep backend skip paths out of the legacy CPU admission branch.

Centrality: Core, tier 1 all-in-one. P1: preserve prompt/token/state identity. P2: eliminate a demonstrated redundant full-prefix copy. P3: reuse existing scoped lookup and admission contracts. P4: retain scope isolation, cross-scope/L2/L3 promotion, and deterministic receipts. Better because warmed same-tenant fan-out avoids needless KV/recurrent copies while exact token outputs remain unchanged.

Scope: internal/agent/inkernel_decode.go and internal/agent/inkernel_redundant_snapshot_test.go only. No model/compute changes or live hardware operations. Device throughput and 80-percent roofline attainment remain unproven by this unit.

Witness: a failing-before production planner regression uses the existing real synthetic model and counting backend, primes one tenant snapshot, then runs two exact requests as different agents of the same tenant. Each exact hit must retain full match/device-L1 identity, produce cold-identical tokens, and clone only once; the cached device snapshot remains resident. Existing scope promotion/isolation and prefix tests remain green. Run in the managed detached worktree through the WSL test runner.

**Comment by anthony-chaudhary** [comment]
Implemented the two-file candidate in a managed detached worktree at base `0faa1c1ef662b274f901fd0bbf7913297adb4635`; not landed yet. The scope returned by scoped snapshot lookup is preserved, and exact device-L1 cached-logit hits already held at the same tenant skip the entire readmission block. Cross-scope hits and L2/L3 restoration retain admission.

Failure-first WSL regression: the second same-tenant agent cloned 12 tensors instead of the 6 needed for lookup. After the fix, the strengthened regression passes for unscoped requests and three agents within one tenant: each of two warm replays clones only 6 tensors, produces the same 3 tokens as cold generation (including two restored-state forward steps), and retains identical device snapshot bytes/tokens. Existing exact-hit, L2, L3, authenticated isolation, and checkpoint cases passed in the same selection (Go 1.26.6, package 0.860 s). `git diff --check` passed.

Witness: `./test.ps1 -count=1 -timeout=90s -run 'TestInKernel(ExactDeviceL1|ReuseExactHit|Backend.*(Prefix|CachedLogits)|AuthenticatedPrefix|SnapshotCheckpoint)' -v ./internal/agent`, with a dedicated worktree-derived ext4 mirror. This runs the real planner on a tiny synthetic hybrid model and recording CPU backend; it proves the copy/ownership behavior, not physical GPU throughput. No 80-percent roofline claim. Independent reviewer is verifying the stable candidate before landing.

**Comment by anthony-chaudhary** [comment]
Resolved and witnessed by TestInKernelSameTenantExactDeviceL1CachedLogitsSkipsRedundantSnapshotClone in commit 7b280579b.
